### PR TITLE
[cmake] Fix single config generators handling

### PR DESCRIPTION
**Pull request by vinograd47**
* Add `USE_BUILD_TYPE_SUBFOLDER` CMake option to append `CMAKE_BUILD_TYPE` to output binary directory. Initialize it to `ON` for UNIX to keep current behavior.
* Remove forced `CMAKE_CONFIGURATION_TYPES` initialization, use user provided value instead.

This will allow to use single config generators (like Ninja) on Windows with MSVC compilers and get binaries in per-config sub-folders in the same way as on UNIX.

### CODE REVIEW

File: ngraph/core/CMakeLists.txt [hunk 1]
```diff
@@ -119,14 +119,12 @@ if (NOT WIN32)
...
-# Build subdirectories for all build types on Windows
-if(WIN32)
-    foreach(BUILD_TYPE Release Debug RelWithDebInfo MinSizeRel)
-        if(NOT EXISTS ${NGRAPH_BUILD_DIR}/${BUILD_TYPE})
-            file(MAKE_DIRECTORY ${NGRAPH_BUILD_DIR}/${BUILD_TYPE})
-        endif()
-    endforeach()
-endif()
+# Create sub-directories for all build types in case of multi-configuration generators
+foreach(BUILD_TYPE IN LISTS CMAKE_CONFIGURATION_TYPES)
```
Comment (ilyachur) [inline ngraph/core/CMakeLists.txt:123]: What if `CMAKE_CONFIGURATION_TYPES` is not defined?

Comment (vinograd47) [reply ngraph/core/CMakeLists.txt:123]: It will be dereferenced as empty list and the loop will be skipped.

Comment (ilyachur) [reply ngraph/core/CMakeLists.txt:123]: Can we just remove this cycle? I think it should affect nothing.

Comment (vinograd47) [reply ngraph/core/CMakeLists.txt]: Removed

**Comment by vinograd47** [comment]
```
C:\a\1\openvino\inference-engine\src\transformations\include\ngraph_ops/type_relaxed.hpp(160): error C4996: 
'ngraph::op::util::FusedOp::validate_and_infer_types': FusedOp approach was deprecated! Please use inheritance from usual Op instead of FusedOp
```

Any ideas why it started to fail?

**Comment by ilya-lavrenov** [comment]
> Any ideas why it started to fail?

https://github.com/openvinotoolkit/openvino/pull/3803

**Review by ilya-lavrenov — APPROVED** [review]

**Pull request merged**